week-3/A-array-find/exercise.js

```diff
@@ -20,7 +20,10 @@ var names = [
   "Ahmed"
 ];
 
-var longNameThatStartsWithA = findLongNameThatStartsWithA(names);
+function findLongNameThatStartsWithA (names){
+  return  names.length > 7 && names[0] === "A";
+}
```

Comment:
This will work OK, but what does this function do (by itself)? Is `findLongNameThatStartsWithA` a good name. What type is the parameter? Is `names` a good name for it?

Reply:
I mean that that functions and parameters should be named to indicate what they do and represent. The function as written checks the length and starting character of a single string and returns true or false (it's a "predicate", which we saw before in the class). It doesn't find anything, so it doesn't make sense to have "find" in the name. And it takes a single name as the parameter. Calling the parameter `names` instead of `name` gives the impression that the parameter contains more than one name (e.g. an array).